docs/DeveloperGuide.md

```diff
@@ -308,6 +336,107 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case resumes at step 2.
 
+**UC03: Add a label to a contact**
+
+**MSS**
+
+1. User requests to list contacts.
+2. d'Intérieur shows a list of contacts.
+3. User requests to add a label to a specific contact in the list.
+4. d'Intérieur adds the label to the contact and shows the contact.
+
+  Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. AddressBook shows an error message.
+
+      Use case resumes at step 2.
+
+* 3b. The label requested does not currently exist in d'Intérieur.
+
+  * To be added
+
+**UC04: Filter contacts using labels**
+
+**MSS**
+
+1. User requests list of contacts filtered to those containing the given label.
+2. d'Intérieur shows a list of contacts who have the given label.
+
+  Use case ends.
+
+**Extensions**
+
+* 1a. The given label does not exist.
+
+  * To be added
+
+**UC05: Adding a note to a contact**
+
+**MSS**
+
+1. User requests to add a note to a specific contact in the list.
+2. d'Intérieur adds the note to the contact and shows the contact.
+
+  Use case ends.
+
+**Extensions**
+
+* 1a. User enters only whitespaces or nothing as a note.
+
+  * d'Intérieur shows an error message that notes cannot be empty.
```

Comment:
Missing extension number, `1a1.`

Additionally, I feel that it might make more sense to not show any error message at all and just ignore the command if the note given is blank, but that is entirely my personal opinion and I don't really feel very strongly about this suggestion.

Reply:
I see, but if the command is ignored, I think it is important to give some feedback to the user that that is the outcome. It does not have to be an error message, perhaps just a normal feedback like "No notes added.".

Reply:
> I see, but if the command is ignored, I think it is important to give some feedback to the user that that is the outcome. It does not have to be an error message, perhaps just a normal feedback like "No notes added.".

That sounds good since some form of user feedback would be useful to indicate system response. On top of that, perhaps we could add a default "To be added.." for blank notes or some form of default comment.

Reply:
ok! Edited to an alert message.